server/application_runner.go

```diff
@@ -297,6 +301,13 @@ func (r *ApplicationRunner) ScheduleApproval(s *Server, pipeline *api.Pipeline)
 		log.Error("failed to get IM setting", zap.String("settingName", string(settingName)), zap.Error(err))
 		return
 	}
+	if setting == nil {
+		log.Error("cannot find IM setting")
```

Comment:
empty IM setting

Reply:
We createIfNotExist empty IM setting on server startup, "cannot find" is more suitable here, IMO.